File: app/src/main/java/com/remotearthsolutions/expensetracker/services/FileProcessingService.java
package com.remotearthsolutions.expensetracker.services;

import android.app.Activity;
import com.remotearthsolutions.expensetracker.databaseutils.models.AccountModel;
import com.remotearthsolutions.expensetracker.databaseutils.models.CategoryModel;
import com.remotearthsolutions.expensetracker.databaseutils.models.ExpenseModel;
import com.remotearthsolutions.expensetracker.databaseutils.models.dtos.CategoryExpense;

import java.util.List;

public interface FileProcessingService {
    void writeOnCsvFile(Activity activity, String content, Runnable runnable);
    List<CategoryExpense> readFromCsvFile(Activity activity);
    void loadTableData(String filepath, Callback callback);
    List<String> getNameOfAllCsvFile();
    void shareFile(Activity activity);

    interface Callback {
        void onComplete(List<CategoryModel> categories, List<ExpenseModel> expenseModels, List<AccountModel> accountModels);
    }
}

File: app/src/main/java/com/remotearthsolutions/expensetracker/services/FileProcessingService.kt
package com.remotearthsolutions.expensetracker.services

import android.app.Activity
import android.net.Uri
import com.remotearthsolutions.expensetracker.databaseutils.models.AccountModel
import com.remotearthsolutions.expensetracker.databaseutils.models.CategoryModel
import com.remotearthsolutions.expensetracker.databaseutils.models.ExpenseModel
import com.remotearthsolutions.expensetracker.databaseutils.models.dtos.CategoryExpense
import java.net.URI

interface FileProcessingService {
    fun writeOnCsvFile(
        activity: Activity,
        content: String?,
        onSuccessRunnable: Runnable?,
        onFailureRunnable: Runnable?
    )

    fun readFromCsvFile(activity: Activity): List<CategoryExpense>?
    fun loadTableData(
        fileURI: Uri?,
        callback: Callback?
    )

    val nameOfAllCsvFile: List<String>?
    fun shareFile(activity: Activity)
    interface Callback {
        fun onComplete(
            categories: List<CategoryModel>?,
            expenseModels: List<ExpenseModel>?,
            accountModels: List<AccountModel>?
        )

        fun onFailure()
    }
}